plugins/systemd/systemd_mem

```diff
@@ -0,0 +1,121 @@
+#!/bin/sh
+: << =cut
+
+=head1 NAME
+
+systemd_mem - Plugin to graph the memory usage of multiple systemd services as reported by systemctl
+
+=head1 CONFIGURATION
+
+The environment variable "services" is used to provide a space separated list of services to graph.
+
+Example:
+   [systemd_mem]
+   env.services apache2 tomcat9 mysql
+
+env.services defaults to "munin-node".
+
+=head1 VERSION
+
+1.0
+
+=head1 AUTHOR
+
+Paul Alexandrow <paul@alexandrow.org>
+
+=head1 LICENSE
+
+Copyright 2021 Paul Alexandrow, paul@alexandrow.org
+
+Permission is hereby granted, free of charge, to any person obtaining a copy of this software
+and associated documentation files (the "Software"), to deal in the Software without
+restriction, including without limitation the rights to use, copy, modify, merge, publish,
+distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the
+Software is furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all copies or
+substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+=head1 MAGIC MARKERS
+
+#%# family=auto
+#%# capabilities=autoconf
+
+=cut
+
+. "$MUNIN_LIBDIR/plugins/plugin.sh"
+
+get_services() {
+    services=${services:-"munin-node"}
+    echo "$services" | awk 'BEGIN{RS=" "}{$1=$1}1'
```

Comment:
What is the purpose of the `awk` filtering here?
Maybe `service=${services:-munin-node}` would be sufficient?

This function feels a bit weird, since it changes its input (as a side-effect due to the non-local variable `services`). Thus multiple invocations of the function *could* (in theory) change the result.

Reply:
I am in no way a bourne-pro, so any help is appreciated. What I am trying to achieve is, to get the space separated plugin setting `apache php munin ...` to be newline separated in order to use it later in one of the for-loops. Because there are two of them, I wanted to put the space-to-newline conversion in a function. Could you kindly propose what would be a better way to achieve that?

Reply:
You can loop any separator in a `for` loop (see `IFS`: *<space>, <tab> and <newline>*). Thus the conversion is probably not necessary at all?

Reply:
Thanks for the hint - i was not aware of `IFS` (and the fact, that `space` is already a default). You were right that the whole part in question could be removed.